functionary/prompt_template/prompt_template_v2.py

```diff
@@ -335,16 +243,21 @@ def update_response_state_from_delta_text(
         finish_reason: Optional[str],
         tool_choice: Any,
     ) -> Tuple[Dict[str, Any], Union[None, Dict, List[Dict]]]:
+        func_name = None
+        response_type = None
+        skip_until_reach = ""
+
         if len(current_state) == 0:  # empty dict, at the first_time
             if tool_choice == "auto":
                 response_type, skip_until_reach = None, self.content_token
                 func_name = None
             elif tool_choice == "none":
                 response_type, skip_until_reach = "text", ""
                 func_name = self.predefined_func_names[PredefinedFuncTypes.no_tool_call]
-            else:
+            elif type(tool_choice) is not str:
```

Comment:
Same problem as above. We should also check if `tool_choice is not None`. I understand currently tool_choice is always either str or Tool class but some other engineers who are not so familiar with the codebase may include None back next time. This check can work if it really happens.

Reply:
Sure